research/cifar10/preprocess.py

```diff
@@ -100,20 +100,20 @@ def preprocess_data(
 
     # Partition train data
     heterogeneous_partitioner = DirichletLabelBasedAllocation(
-        number_of_partitions=num_clients, unique_labels=list(range(10)), beta=beta, min_label_examples=2
+        number_of_partitions=num_clients, unique_labels=list(range(10)), beta=beta, min_label_examples=1
     )
     train_partitioned_datasets, train_partitioned_dist = heterogeneous_partitioner.partition_dataset(
-        training_set, max_retries=5
+        training_set, max_retries=-1
```

Comment:
What does `-1` retries mean?

Reply:
I'll put a note in the documentation. It effectively means infinite. I'm kind of "abusing" how the underlying code works to get this functionality.

Reply:
I think it's fine to use it like that, having it in the documentation helps understanding the use so I think we should do it.